### PR TITLE
Cleanup JavaParser.annotation

### PR DESCRIPTION
**Pull request by dwijnand**
I accidentally thought a reported bug was here while it was actually in
Typer, so this is just some cleanups (and a TODO) in the area.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/javac/JavaParsers.scala [hunk 1]
```diff
@@ -379,101 +379,60 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
...
-                }
-              }
-            case _ if in.token == IDENTIFIER =>
-              qualId(orClassLiteral = true)
-            case _ =>
-              in.nextToken()
-              EmptyTree
-          }
+    def annotation(): Tree = atPos(in.pos) {
+      object LiteralK { def unapply(token: Token) = tryLiteral() }
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/javac/JavaParsers.scala:383]: hmm, an extractor that advances the token stream as a side effect is maybe a little wild...?

File: src/compiler/scala/tools/nsc/javac/JavaParsers.scala [hunk 2]
```diff
@@ -379,101 +379,60 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
...
-                  }
-                }
-              }
-            case _ if in.token == IDENTIFIER =>
-              qualId(orClassLiteral = true)
-            case _ =>
-              in.nextToken()
-              EmptyTree
-          }
+    def annotation(): Tree = atPos(in.pos) {
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/javac/JavaParsers.scala]: it took me a while to spot the `atPos` - i feel that it's more clearly placed down at the expression

**Review by lrytz — COMMENTED** [review]
Very very nice cleanup! Just two small remarks.

File: src/compiler/scala/tools/nsc/javac/JavaParsers.scala (hunk 2 above)
Comment (dwijnand) [reply src/compiler/scala/tools/nsc/javac/JavaParsers.scala]: Sure, np.

File: src/compiler/scala/tools/nsc/javac/JavaParsers.scala (hunk 1 above)
Comment (dwijnand) [reply src/compiler/scala/tools/nsc/javac/JavaParsers.scala:383]: Good point...  IMO the case alignment of `elementValue` is too nice to lose, so I'll try changing `tryLiteral` and its (2) usages.

**Pull request merged**